### PR TITLE
chore: enable eslint-plugin-perfectionist on rule-tester package

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
## PR Checklist

- [x] Addresses an existing open issue: continues #8479
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview

Enables sorting on `packages/rule-tester`.

💖

### CODE REVIEW

File: eslint.config.mjs [hunk 1]
```diff
@@ -578,7 +578,10 @@ export default tseslint.config(
   },
   {
     extends: [perfectionistPlugin.configs['recommended-alphabetical']],
-    files: ['packages/utils/src/**/*.ts'],
+    files: [
+      'packages/rule-tester/{src,tests,typing}/**/*.ts',
```
Comment (JoshuaKGoldberg) [inline eslint.config.mjs:582]:
```suggestion
      'packages/rule-tester/{src,tests,typings}/**/*.ts',
```

File: packages/rule-tester/src/types/DependencyConstraint.ts [hunk 2]
```diff
@@ -6,12 +6,12 @@ export interface RangeOptions {
...
 
 export interface SemverVersionConstraint {
-  readonly range: string;
   readonly options?: RangeOptions | boolean;
+  readonly range: string;
 }
 export type AtLeastVersionConstraint =
-  | `${number}.${number}.${number}-${string}`
   | `${number}.${number}.${number}`
+  | `${number}.${number}.${number}-${string}`
```
Comment (bradzacher) [inline packages/rule-tester/src/types/DependencyConstraint.ts:14]: weird that this sorted this way.

Why does `-` come AFTER "nothing", but `.` comes BEFORE nothing.
Is this a bug? What the what?

File: packages/rule-tester/src/utils/config-validator.ts [hunk 3]
```diff
@@ -25,8 +26,8 @@ const ruleValidators = new WeakMap<AnyRuleModule, ValidateFunction>();
 let validateSchema: ValidateFunction | undefined;
 const severityMap = {
   error: 2,
-  warn: 1,
   off: 0,
+  warn: 1,
```
Comment (bradzacher) [inline packages/rule-tester/src/utils/config-validator.ts:30]: this is another case where the value ordering is better than the prop name ordering

File: packages/rule-tester/src/utils/flat-config-schema.ts [hunk 4]
```diff
@@ -10,17 +10,17 @@ import { normalizeSeverityToNumber } from './severity';
...
 
 const ruleSeverities = new Map<SharedConfig.RuleLevel, SharedConfig.Severity>([
-  [0, 0],
+  ['error', 2],
   ['off', 0],
-  [1, 1],
   ['warn', 1],
+  [0, 0],
+  [1, 1],
   [2, 2],
```
Comment (bradzacher) [inline packages/rule-tester/src/utils/flat-config-schema.ts:23]: this is another case where the value ordering is better than the prop name ordering

**Review by bradzacher — DISMISSED** [review]

File: packages/rule-tester/src/types/DependencyConstraint.ts (hunk 2 above)
Comment (JoshuaKGoldberg) [reply packages/rule-tester/src/types/DependencyConstraint.ts:14]: I think it's that <code>`</code> is being treated as before `-` alphabetically?

Comment (bradzacher) [reply packages/rule-tester/src/types/DependencyConstraint.ts:14]: That must be it - it must be considering the quote character as part of the thing.

Would be worth filing a bug for that.
"if the two types are the same literal type (eg both template literal types) then the quote character should be excluded from the sort comparison.

Comment (JoshuaKGoldberg) [reply packages/rule-tester/src/types/DependencyConstraint.ts:14]: 👍 noted in https://github.com/typescript-eslint/typescript-eslint/issues/8479#issuecomment-2307391859

**Pull request merged**